### DISCUSSION TITLE
The binary version of its metadata is 1.9.0, expected version is 1.7.1.  FAILURE: Build failed with an exception.

### DISCUSSION THREAD

**Discussion by ronyarbiv** [airsdk/Adobe-Runtime-Support#3188] [Q&A]
Hi,

I am using latest Air sdk and latest Distriqt Anes (ads,firebase,rate,dialog,share,inapp...) {@marchbold @distriqt-developer}
Getting this error on build. 

any suggestions?

Unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: ERROR:D8: com.android.tools.r8.kotlin.H
e: C:/Users/dnr/AppData/Local/Temp/57f1f2cf-112a-4b43-b1c2-9e0b74dfd959/app/libs/asset-delivery-ktx-2.2.1.jar!/META-INF/java.com.google.android.apps.play.store.sdk.playcore.ktx_playcore_asset_delivery_ktx.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

FAILURE: Build failed with an exception.

**Comment by ronyarbiv** [comment]
The error you're encountering during the build process indicates a Kotlin version incompatibility issue with one of your dependencies, specifically with asset-delivery-ktx-2.2.1.jar. The error message states that this module was compiled with Kotlin version 1.9.0, but your project expects version 1.7.1

Does anyone know how can i fix that? asset-delivery-ktx-2.2.1.jar is that file in the sdk or at any of the Anes?

**Reply by ronyarbiv** [reply]
It's in com.google.android.play.ane 
May you please fix that?
@marchbold @distriqt-developer

**Reply by marchbold** [reply]
Can you check what version of the Android build tools you have installed and make sure it's updated. The latest android libs use kotlin > 1.7.1 so you will need to update or roll back to the previous release of the extensions.

**Reply by ronyarbiv** [reply]
Hi,

Thank you.

When i switch the specific Ane google. android.play (from distriqt) with the oldest i had from August, it works. I am using android 33.
Is it ok to work with this oldest, and other new? I have seen that it is only using for rating.

**Reply by marchbold** [reply]
The `com.google.android.play` extension is used by several other extensions. You'll need to ensure you are using the older version of them as well. This includes InAppBilling, ApplicationRater and Harman's PlayAssetDelivery

**Reply by ronyarbiv** [reply]
Do you have control on the latest com.android.google.play i have took it from distriqt github. May yo please check Is it working with 1.7.1 and not 1.9?
Hope that's ok that i am asking, i dont understand much on anes and use all of my anes from distriqt.

**Reply by marchbold** [reply]
Hi, In order to access the latest functionality we need to use the latest kotlin dependencies. So something is amiss in your build environment as you should be able to use kotlin > 1.7. Try installing v34 of the build tools.

**Reply by ronyarbiv** [reply]
Thanks for the directions, i will try and do it.

**Reply by marchbold** [reply]
What have you got your android minimum sdk version set to in your manifest additions? Try increasing it to 25+.

**Reply by ronyarbiv** [reply]
Thank you.

Setting the minimum SDK to 24 resolved the issue; however, versions 23 and lower still yield the same error. Notably, 10% of my users are on SDK versions 21, 22, and 23. Is there an alternative fix we might consider? There is an open thread, #3197—perhaps we should continue the discussion there?
